utils/misc/dbutils.go

```diff
@@ -21,11 +22,27 @@ func GetConnectionString(c *config.Config) string {
 	// Application Name can be any string of less than NAMEDATALEN characters (64 characters in a standard PostgreSQL build).
 	// There is no need to truncate the string on our own though since PostgreSQL auto-truncates this identifier and issues a relevant notice if necessary.
 	appName := DefaultString("rudder-server").OnError(os.Hostname())
+	if len(componentName) > 0 {
+		appName = fmt.Sprintf("%s-%s", componentName, appName)
+	}
 	return fmt.Sprintf("host=%s port=%d user=%s "+
 		"password=%s dbname=%s sslmode=%s application_name=%s",
 		host, port, user, password, dbname, sslmode, appName)
 }
 
+// SetApplicationNameInDBDNS sets application name in db connection url
+// if application name is already present in dns it will get override by the appName
+func SetApplicationNameInDBConnectionURL(connectionUrl, appName string) (string, error) {
+	connUrl, err := url.Parse(connectionUrl)
+	if err != nil {
+		return "", err
+	}
+	queryParams := connUrl.Query()
+	queryParams.Set("application_name", appName)
+	connUrl.RawQuery = queryParams.Encode()
+	return connUrl.String(), nil
+}
```

Comment:
We can probably use a shorter function name here.

```suggestion
// SetAppNameInDBConnURL sets application name in db connection url
// if application name is already present in dns it will get override by the appName
func SetAppNameInDBConnURL(connectionURL, appName string) (string, error) {
	connURL, err := url.Parse(connectionURL)
	if err != nil {
		return "", fmt.Errorf("parsing connection url: %w", err)
	}

	queryParams := connURL.Query()
	queryParams.Set("application_name", appName)
	connURL.RawQuery = queryParams.Encode()
	return connURL.String(), nil
}
```